src/Analyser/MutatingScope.php

```diff
@@ -507,7 +507,7 @@ public function getVariableType(string $variableName): Type
 		}
 
 		if ($this->isGlobalVariable($variableName)) {
-			return new ArrayType(new StringType(), new MixedType($this->explicitMixedForGlobalVariables));
+			return new ArrayType(TypeCombinator::union(new IntegerType(), new StringType()), new MixedType($this->explicitMixedForGlobalVariables));
```

Comment:
I'd like more if this was benevolent union type. It's annoying to require users to handle `int` if it can never happen inside their app. You can do `new BenevolentUnionType([new IntegerType(), new StringType()])`.

More about this: https://phpstan.org/config-reference#checkbenevolentuniontypes